src/Model/Resolver/Products/DataPostProcessor/Stocks.php

```diff
@@ -142,17 +198,40 @@ public function process(
         $formattedStocks = [];
 
         foreach ($stockItems as $stockItem) {
-            $inStock = $stockItem->getStatus() === SourceItemInterface::STATUS_IN_STOCK;
-
             $leftInStock = null;
             $qty = $stockItem->getQuantity();
+            $sku = $stockItem->getSku();
+
+            $inStock = (($stockItem->getStatus() === SourceItemInterface::STATUS_IN_STOCK)
+                        and $qty > 0)? true : false;
+
+            if ($inStock) {
+                $productSalableQty = $this->getProductSalableQty->execute($sku, $stockId);
```

Comment:
Keep in mind that this will execute a number of DB requests inside, for each of the loaded products, separately.

This defeats the purpose of having a performance post processor for stocks, as we want to load grouped data whenever possible.